### PR TITLE
chore: migrate to cdklabs projen project type

### PR DESCRIPTION
**Pull request by kaizencc**
Reference: https://github.com/cdklabs/cdk-ops/issues/2581

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,6 +22,10 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.18.0
```
Comment (kaizencc) [inline .github/workflows/build.yml:28]: do we want to go ahead and enforce node-version: 16 or wait for us to make that change in cdklabs-projen-project-types (my preference)?

Comment (mrgrain) [reply .github/workflows/build.yml:28]: Some packages already fail so you might as well.

But note that there's a difference between enforcing it and stop testing for it.
Personally I'd rather do the latter and only enforce it in a few months or even only when we actually want to use certain language features.

File: .npmignore
Comment (mrgrain) [inline .npmignore:1]: Removing this is unexpected

File: .projenrc.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
-const { typescript } = require('projen');
+import { CdklabsTypeScriptProject } from 'cdklabs-projen-project-types';
 
-const project = new typescript.TypeScriptProject({
+const project = new CdklabsTypeScriptProject({
+  projenrcTs: true,
```
Comment (mrgrain) [inline .projenrc.ts:4]: Maybe we should change the default in `CdklabsTypeScriptProject`?

File: package.json [hunk 3]
```diff
@@ -104,5 +110,6 @@
     }
   },
   "types": "lib/index.d.ts",
-  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"npx projen\"."
+  "private": true,
```
Comment (mrgrain) [inline package.json:113]:
```suggestion
  "private": false,
```

File: package.json [hunk 4]
```diff
@@ -22,24 +22,26 @@
...
+    "test:update": "npx projen test:update",
     "test:watch": "npx projen test:watch",
     "unbump": "npx projen unbump",
     "upgrade": "npx projen upgrade",
     "watch": "npx projen watch",
     "projen": "npx projen"
   },
   "author": {
-    "name": "Amazon Web Servies",
+    "name": "Amazon Web Services",
```
Comment (mrgrain) [inline package.json:33]: lol 😹

File: package.json [hunk 5]
```diff
@@ -60,6 +63,9 @@
     "jsii-rosetta": "^1.80.0",
     "yargs": "^17.7.1"
   },
+  "engines": {
+    "node": ">= 14.18.0"
+  },
```
Comment (mrgrain) [inline package.json:68]: Starting to wonder if we should not enforce engines at all by default. Especially for these smaller side projects, it seems unlikely people will be start complaining about `cdk-generate-synthetic-examples` not working with node10

File: tsconfig.dev.json [hunk 6]
```diff
@@ -28,7 +28,9 @@
   "include": [
     ".projenrc.js",
```
Comment (mrgrain) [inline tsconfig.dev.json:29]: Ugh. This config is still not clean in projen upstream 😓

**Review by mrgrain — CHANGES_REQUESTED** [review]

**Review by mrgrain — APPROVED** [review]

**Pull request merged**